Feat: Add a `yarn dev` cmd for local dev without Docker (#6170)
circleci/circleci-docs: docs/local-development.md

**Note:** In the event you find yourself needing to cleanup docker/jekyll cache, you can use the `yarn clean` command.

## 2. Local Development with Ruby and Bundler

> If you already have a stable Ruby environment (currently Ruby 2.7.4) and feel comfortable installing external dependencies, you can follow these instructions. Check out the [Gemfile](https://github.com/circleci/circleci-docs/blob/master/Gemfile) for the Ruby version we're currently using. We recommend [RVM](https://rvm.io/) for managing multiple Ruby versions.

Reviewer: ```suggestion
If you already have a stable Ruby environment (currently Ruby 2.7.4) and feel comfortable installing external dependencies, you can follow these instructions. Check out the [Gemfile](https://github.com/circleci/circleci-docs/blob/master/Gemfile) for the Ruby version we are currently using. We recommend [RVM](https://rvm.io/) for managing multiple Ruby versions.
```

1. Install Docker for your platform: <https://docs.docker.com/engine/install/>
2. Clone the CircleCI docs repo: `git clone https://github.com/circleci/circleci-docs.git`
3. Start Docker Desktop
4. Add the following line to your `/etc/hosts` file:
   ```bash
   127.0.0.1 ui.circleci.com
   ```
5. Run `bundle install`. You will need to have [Bundler](https://bundler.io/) installed on your local machine.
6. Run `yarn install`. You will need to have [Yarn](https://yarnpkg.com/getting-started) installed on your local machine.
7. Finally, run `yarn dev` to start the site.

> 8. The docs site will now be running on <https://ui.circleci.com/docs/>. If the browser presents to you an HSTS Security Warning, you can safely bypass it as it is an expected outcome of running the Caddy Reverse Proxy in Docker.

Reviewer: ```suggestion
8. The docs site will now be running on <https://ui.circleci.com/docs/>. If the browser displays a HSTS Security Warning, you can safely bypass it, as it is an expected outcome of running the Caddy Reverse Proxy in Docker.
```
